From 8aec01a4d5d098430cdb369a69a85cdbf20ebc99 Mon Sep 17 00:00:00 2001
From: Julien Cristau <jcristau@debian.org>
Date: Thu, 25 Aug 2011 23:08:05 +0200
Subject: [PATCH] Work around the lack of O_CLOEXEC on kbsd

Not that dri2 is going to work there, but well.
---
 src/egl/drivers/dri2/platform_x11.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

Index: mesa/src/egl/drivers/dri2/platform_x11.c
===================================================================
--- mesa.orig/src/egl/drivers/dri2/platform_x11.c	2012-01-18 16:04:58.294653499 +0200
+++ mesa/src/egl/drivers/dri2/platform_x11.c	2012-01-18 16:05:08.790968112 +0200
@@ -1083,7 +1083,13 @@
    if (!dri2_load_driver(disp))
       goto cleanup_conn;
 
+#ifdef O_CLOEXEC
    dri2_dpy->fd = open(dri2_dpy->device_name, O_RDWR | O_CLOEXEC);
+#else
+   dri2_dpy->fd = open(dri2_dpy->device_name, O_RDWR);
+   if (dri2_dpy->fd >= 0)
+      fcntl(dri2_dpy->fd, F_SETFD, fcntl(dri2_dpy->fd, F_GETFD) | FD_CLOEXEC);
+#endif
    if (dri2_dpy->fd == -1) {
       _eglLog(_EGL_WARNING,
 	      "DRI2: could not open %s (%s)", dri2_dpy->device_name,
